lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -671,12 +685,23 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
 
         alert_rules_as_dict = alert_rules.as_dict()
 
+        if extra_alerts_list:
+            if alert_rules_as_dict:
```

Comment:
Did we decide against something like:

```python
alert_rules_as_dict["groups"] = alert_rules_as_dict.get("groups", []) + extra_alerts_list
```

This seems to fulfill all of the requirements without the conditionals.

Reply:
in the case if `alert_rules_as_dict` and `extra_alerts_list` we current workflow the relation databag will be empty. We will not create `alert_rules` key.

While with proposed implementation databag key will always exist, but will be in some cases empty